scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,323 @@
+    else:
+        doption = np.array((delta, eta, anorm), dtype=typ.lower())
+
+    ioption = np.array((int(bool(cgs)), int(bool(elr))), dtype='i')
+
+    # Determine lwork & liwork:
+    # the required lengths are specified in the PROPACK documentation
+    # BLAS-3 blocking size is 16, but docs don't specify; it's almost surely a
+    # power of 2.
+    blocksize = max(2, int(blocksize))
+    if compute_u or compute_v:
+        lwork = m + n + 9*kmax + 5*kmax*kmax + 4 + max(3*kmax*kmax + 4*kmax + 4,
+                                                       blocksize*max(m, n))
+        liwork = 8*kmax
+    else:
+        lwork = m + n + 9*kmax + 2*kmax*kmax + 4 + max(m + n, 4*kmax + 4)
+        liwork = 2*kmax + 1
+    work = np.empty(lwork, dtype=typ.lower())
+    iwork = np.empty(liwork, dtype=np.int32)
+
+    # dummy arguments: these are passed to aprod, and not used in this wrapper
+    dparm = np.empty(1, dtype=typ.lower())
+    iparm = np.empty(1, dtype=np.int32)
+
+    if typ.isupper():
+        # PROPACK documentation is unclear on the required length of zwork.
+        # Use the same length Julia's wrapper uses
+        # see https://github.com/JuliaSmoothOptimizers/PROPACK.jl/
+        zwork = np.empty(m + n + 32*m, dtype=typ)
+
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, zwork,
+                                                iwork, doption, ioption,
+                                                dparm, iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, zwork, iwork, doption,
+                                            ioption, dparm, iparm)
+    else:
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, iwork,
+                                                doption, ioption, dparm,
+                                                iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, iwork, doption,
+                                            ioption, dparm, iparm)
+
+    if info > 0:
+        raise LinAlgError(
+            f"An invariant subspace of dimension {info} was found.")
+    if info < 0:
+        raise LinAlgError(
+            f"k={k} singular triplets did not converge within "
+            f"kmax={kmax} iterations")
+
+    # catch corner case where A is all zeros and v is not orthogonal
```

Comment:
This felt like a hacky fix to pass an arpack test, but probably shouldn't return all-zero vh?

Reply:
I don't mind it. Seems reasonable that it should be orthogonal, and the choice is not unique, so this is as valid as any.